ee/server/service/mdm.go

```diff
@@ -600,13 +585,11 @@ func (svc *Service) getAutomaticEnrollmentProfile(ctx context.Context) (*fleet.M
 	//
 	// TODO: this will change after #10995 where there can be a DEP profile
 	// per team.
-	var depProf *fleet.MDMAppleEnrollmentProfile
-	for _, prof := range profiles {
-		if prof.Type == "automatic" {
-			depProf = prof
-			break
-		}
-	}
 
-	return depProf, nil
+	// TODO(mna): what needs to happen here? This gets called from initiate SSO, but also
+	// from mdmAppleSyncDEPProfile, apparently when SSO settings are changed? Should this
+	// return the team/no team profile of the user, and only fallback on the default one
+	// if there is none? What if user is in multiple teams?
```

Comment:
Question for @roperzh probably?

Reply:
@mna

1. This gets called in the SSO callback just to grab a `mdm_apple_enrollment_profiles.token`, any token really. At this point whatever is in the JSON profile doesn't matter, the profile has been assigned to the device in ABM and we're just using the token to gate access to the XML profile.
2. This also gets called in `mdmAppleSyncDEPProfile`, which is called in `PATCH /config` if:
  1. SSO is turned on/off, because if SSO is enabled, `configuration_web_url` in the DEP profile needs to be updated.
  2. `AppConfig.ServerSettings.ServerURL` changed, because we need to update `url` and `configuration_web_url` in the profile.

I hope it makes sense, I think we can change `1` as we see fit, I used the tools we had available at the time!

Reply:
Gotcha, thanks for the clarification @roperzh !

Reply:
@roperzh  Basically for 2), if the SSO is turned on/off or server URL changes, all profiles need to be updated, so I think this could be a case to enqueue a new worker job that would take care of re-defining and re-assigning all of those (do they need to be re-assigned or just re-defined?).

Reply:
@mna that sounds good! I think you need to:

1. For each existing profile `x`, redefine a new profile `x'`
2. For each device that had `x` assigned, assign `x'`

Reply:
(this is because with your changes we now have multiple profiles co-existing)

Reply:
@mna sorry, I was trying to be concise and I failed 😅 .

>  When you say create a new profile, you mean call DefineProfile API again with the updated profile, and that will generate a new profile UUID?

Yes, exactly! and with this new UUID you can call AssignProfile again for all devices. There's a bit of synchronization at play here between whatever we store in our DB and Apple's servers.

As a note of color, in my tests, Apple is smart enough to not create a new profile UUID if the JSON you're sending in `DefineProfile` is the same of a profile you have defined before.